backend/store/etcd/silenced_store.go

```diff
@@ -29,48 +29,56 @@ func GetSilencedPath(ctx context.Context, name string) string {
 	return silencedKeyBuilder.WithContext(ctx).Build(name)
 }
 
-// DeleteSilencedEntryByName a silenced entry by its name (subscription + checkname)
-func (s *Store) DeleteSilencedEntryByName(ctx context.Context, silencedName string) error {
-	if silencedName == "" {
-		return errors.New("must specify name")
+// DeleteSilencedEntryByName deletes one or more silenced entries by name
+func (s *Store) DeleteSilencedEntryByName(ctx context.Context, silencedNames ...string) error {
+	if len(silencedNames) == 0 {
+		return nil
+	}
+	ops := make([]clientv3.Op, 0, len(silencedNames))
+	for _, silenced := range silencedNames {
+		ops = append(ops, clientv3.OpDelete(GetSilencedPath(ctx, silenced)))
 	}
 
-	_, err := s.client.Delete(ctx, GetSilencedPath(ctx, silencedName))
-	return err
+	_, err := s.client.Txn(ctx).Then(ops...).Commit()
+	if err != nil {
+		return fmt.Errorf("error deleting silenced entries: %s", err)
+	}
+	return nil
 }
 
 // GetSilencedEntries gets all silenced entries.
-func (s *Store) GetSilencedEntries(ctx context.Context) ([]*types.Silenced, error) {
+func (s *Store) GetSilencedEntries(ctx context.Context) ([]*corev2.Silenced, error) {
 	resp, err := s.client.Get(ctx, GetSilencedPath(ctx, ""), clientv3.WithPrefix())
 	if err != nil {
 		return nil, err
 	}
-	silencedArray, err := s.arraySilencedEntries(resp)
+	silencedArray, err := s.arraySilencedEntries(ctx, resp)
 	if err != nil {
 		return nil, err
 	}
 	return silencedArray, nil
 }
 
-// GetSilencedEntriesBySubscription gets all silenced entries that match a subscription.
-func (s *Store) GetSilencedEntriesBySubscription(ctx context.Context, subscription string) ([]*types.Silenced, error) {
-	if subscription == "" {
-		return nil, errors.New("must specify subscription")
+// GetSilencedEntriesBySubscription gets all silenced entries that match set of subscriptions.
```

Comment:
Nitpick.. totally don't need to do this lol
```suggestion
// GetSilencedEntriesBySubscription gets all silenced entries that match a set of subscriptions.
```